src/gf128.rs

```diff
@@ -12,21 +12,21 @@ bytes!(Key, BLOCKSIZE);
 bytes!(Tag, BLOCKSIZE);
 
 // TODO: Use a 128-bit uint_n instead?
-type Element = u128;
-const IRRED: Element = 0xE1000000000000000000000000000000;
+type Element = U128;
+const IRRED: Element = U128(0xE1000000000000000000000000000000);
 
 fn fadd(x: Element, y: Element) -> Element {
     x ^ y
 }
 
 fn fmul(x: Element, y: Element) -> Element {
-    let mut res: Element = 0;
+    let mut res: Element = U128(0);
     let mut sh = x;
     for i in 0..128 {
-        if y & (1 << (127 - i)) != 0 {
+        if y & (U128(1) << (127 - i)) != U128(0) {
             res ^= sh;
         }
-        if sh & 1 != 0 {
+        if sh & U128(1) != U128(0) {
```

Comment:
Note that this is not constant-time, and is signalled as such by the `secret_integers` crate : 

```rust
error[E0369]: binary operation `!=` cannot be applied to type `secret_integers::U128`
  --> src/gf128.rs:26:39
   |
26 |         if y & (U128(1) << (127 - i)) != U128(0) {
   |            -------------------------- ^^ ------- secret_integers::U128
   |            |
   |            secret_integers::U128
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `secret_integers::U128`

error[E0369]: binary operation `!=` cannot be applied to type `secret_integers::U128`
  --> src/gf128.rs:29:25
   |
29 |         if sh & U128(1) != U128(0) {
   |            ------------ ^^ ------- secret_integers::U128
   |            |
   |            secret_integers::U128
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `secret_integers::U128`

error[E0369]: binary operation `!=` cannot be applied to type `secret_integers::U128`
  --> src/gf128.rs:26:39
   |
26 |         if y & (U128(1) << (127 - i)) != U128(0) {
   |            -------------------------- ^^ ------- secret_integers::U128
   |            |
   |            secret_integers::U128
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `secret_integers::U128`

error[E0369]: binary operation `!=` cannot be applied to type `secret_integers::U128`
  --> src/gf128.rs:29:25
   |
29 |         if sh & U128(1) != U128(0) {
   |            ------------ ^^ ------- secret_integers::U128
   |            |
   |            secret_integers::U128
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `secret_integers::U128`

```